pkg/memory-monitor/src/statistics-collector/main.c

```diff
@@ -0,0 +1,108 @@
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <unistd.h>
+
+#define BUFFER_SIZE 256
+#define OUTPUT_FILE "psi_stats.txt"
+#define DEVICE_OUTPUT_DIR "/persist/memory-monitor/output"
+#define PROC_FILE "/proc/pressure/memory"
+
+void write_to_file(int fd, const char *buffer) {
+    if (write(fd, buffer, strlen(buffer)) == -1) {
+        perror("write");
+        exit(EXIT_FAILURE);
+    }
+    if (fsync(fd) == -1) {
+        perror("fsync");
+        exit(EXIT_FAILURE);
+    }
+}
+
+static void daemonize(void)
+{
+    int pid, sid;
```

Comment:
```suggestion
    int sid;
    pid_t pid;
```

Reply:
Do we have some code standard for that kind of things? If so, please share it so I can check before making PRs.

Reply:
`pid_t` comes from `man 2 fork`.

Reply:
Aaaah, I see, sorry